src/stores/external-event.ts

```diff
@@ -0,0 +1,86 @@
+import { derived, writable, type Writable } from 'svelte/store';
+import type { ExternalEventDB, ExternalEventType } from '../types/external-event';
+import gql from '../utilities/gql';
+import { getRowIdExternalEvent } from '../utilities/hash';
+import { selectedPlanDerivationGroupNames } from './external-source';
+import { gqlSubscribable } from './subscribable';
+import { viewUpdateGrid } from './views';
+
+/* Writeable. */
+export const creatingExternalEventType: Writable<boolean> = writable(false);
+export const createExternalEventTypeError: Writable<string | null> = writable(null);
+
+/* Subscriptions. */
+export const externalEventsDB = gqlSubscribable<ExternalEventDB[]>(
+  gql.SUB_PLAN_EXTERNAL_EVENTS_DG,
+  { derivation_group_names: selectedPlanDerivationGroupNames },
+  [],
+  null,
+  transformExternalEvents,
+);
+export const externalEventTypes = gqlSubscribable<ExternalEventType[]>(gql.SUB_EXTERNAL_EVENT_TYPES, {}, [], null);
+
+// use to track which event is selected in the plan view, as this information is shared across several sibling panels
+export const selectedExternalEventId: Writable<number | null> = writable(null);
+
+/* Derived. */
+export const selectedExternalEvent = derived(
+  [selectedExternalEventId, externalEventsDB],
+  ([$selectedExternalEventId, $externalEventsDB]) => {
+    const selected = $externalEventsDB.find(ee => getRowIdExternalEvent(ee.pkey) === $selectedExternalEventId);
```

Comment:
Can we create a map of external events so that we don't have to loop through each event to find it? Similar to how we do it for activity directives.